### ISSUE TITLE
`Time::checked_add_f64` and `Time::checked_sub_f64` panic on too large input

### ISSUE THREAD

**Issue by lopopolo** [artichoke/artichoke#2194]
```rust
use core::time::Duration;

fn main() {
    dbg!(Duration::from_secs_f64(f64::MAX));
}
```

Output:

```
Compiling playground v0.0.1 (/playground)
    Finished dev [unoptimized + debuginfo] target(s) in 0.58s
     Running `target/debug/playground`
thread 'main' panicked at 'can not convert float seconds to Duration: value is either too big or NaN', /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/time.rs:744:23
note: [run with `RUST_BACKTRACE=1` environment variable to display a backtrace](https://play.rust-lang.org/#)
```

There is an unstable API, `Duration::try_from_secs_{f32, f64}` which will allow for a checked conversion. See:

- https://github.com/rust-lang/rust/issues/83400

Once this API is stabilized, convert this code to use the fallible conversion:

https://github.com/artichoke/artichoke/blob/2db53033309c3e4f923fdea960d0884c4b47d1c8/spinoso-time/src/time/tzrs/math.rs#L163-L167

https://github.com/artichoke/artichoke/blob/2db53033309c3e4f923fdea960d0884c4b47d1c8/spinoso-time/src/time/tzrs/math.rs#L244-L248

**Comment by lopopolo** [comment]
I've put up a stabilization PR for this feature in Rust `std` here: https://github.com/rust-lang/rust/pull/102271.

**Comment by lopopolo** [comment]
These functions were stabilized and released today in Rust 1.66.0:

- https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.try_from_secs_f32
- https://blog.rust-lang.org/2022/12/15/Rust-1.66.0.html#stabilized-apis
